modules/weather/apixu/apixu_test.go

```diff
@@ -0,0 +1,150 @@
+		{"1153", "Light drizzle", weather.Drizzle},
+		{"1168", "Freezing drizzle", weather.Drizzle},
+		{"1171", "Heavy freezing drizzle", weather.Drizzle},
+		{"1180", "Patchy light rain", weather.Rain},
+		{"1183", "Light rain", weather.Rain},
+		{"1186", "Moderate rain at times", weather.Rain},
+		{"1189", "Moderate rain", weather.Rain},
+		{"1192", "Heavy rain at times", weather.Rain},
+		{"1195", "Heavy rain", weather.Rain},
+		{"1198", "Light freezing rain", weather.Rain},
+		{"1201", "Moderate or heavy freezing rain", weather.Rain},
+		{"1204", "Light sleet", weather.Sleet},
+		{"1207", "Moderate or heavy sleet", weather.Sleet},
+		{"1210", "Patchy light snow", weather.Snow},
+		{"1213", "Light snow", weather.Snow},
+		{"1216", "Patchy moderate snow", weather.Snow},
+		{"1219", "Moderate snow", weather.Snow},
+		{"1222", "Patchy heavy snow", weather.Snow},
+		{"1225", "Heavy snow", weather.Snow},
+		{"1237", "Ice pellets", weather.Hail},
+		{"1240", "Light rain shower", weather.Rain},
+		{"1243", "Moderate or heavy rain shower", weather.Rain},
+		{"1246", "Torrential rain shower", weather.Rain},
+		{"1249", "Light sleet showers", weather.Sleet},
+		{"1252", "Moderate or heavy sleet showers", weather.Sleet},
+		{"1255", "Light snow showers", weather.Snow},
+		{"1258", "Moderate or heavy snow showers", weather.Snow},
+		{"1261", "Light showers of ice pellets", weather.Hail},
+		{"1264", "Moderate or heavy showers of ice pellets", weather.Hail},
+		{"1273", "Patchy light rain with thunder", weather.Thunderstorm},
+		{"1276", "Moderate or heavy rain with thunder", weather.Thunderstorm},
+		{"1279", "Patchy light snow with thunder", weather.Snow},
+		{"1282", "Moderate or heavy snow with thunder", weather.Snow},
+		// Unknown condition.
+		{"0", "unknown", weather.ConditionUnknown},
+	} {
+		wthr, _ := Provider(ts.URL + "/tpl/conditions.json?code=" + tc.apixuCondition).GetWeather()
+		require.Equal(t, tc.expected, wthr.Condition,
+			"Apixu %s (%s)", tc.description, tc.apixuCondition)
+	}
+}
+
+func TestProviderBuilder(t *testing.T) {
+	for _, tc := range []struct {
+		expected    string
+		actual      weather.Provider
+		description string
+	}{
+		{"key=foo&q=29617", New("foo").Query("29617"), "Zip code"},
+		{"key=foo&q=Paris", New("foo").Query("Paris"), "City name"},
+		{"key=foo&q=145.77%2C-16.92", New("foo").Query("145.77,-16.92"), "Latitude and Longitude"},
+		{"key=foo&q=metar%3AEGLL", New("foo").Query("metar:EGLL"), "METAR"},
+		{"key=foo&q=iata%3ADXB", New("foo").Query("iata:DXB"), "IATA"},
+		{"key=foo&q=100.0.0.1", New("foo").Query("100.0.0.1"), "IP lookup"},
+		{"key=foo&q=auto%3Aip", New("foo").Query("auto:ip"), "Auto IP lookup"},
+	} {
+		expected := "http://api.apixu.com/v1/current.json?" + tc.expected
+		require.Equal(t, expected, string(tc.actual.(Provider)), tc.description)
+	}
+}
+
+func TestLive(t *testing.T) {
+	cron.Test(t, func() error {
+		wthr, err := New(os.Getenv("APIXU_API_KEY")).
```

Comment:
Oh and for cron tests, I use the "WEATHER_$PROVIDER_API_KEY" format, so that the testonly api keys show up together in the settings.

Reply:
Makes sense. I'll change that as well.